lib/ui/window.dart

```diff
@@ -150,8 +156,30 @@ class Locale {
   /// Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).
   /// Typically this means the primary language subtag should be lowercase and
   /// the region subtag should be uppercase.
+  ///
+  /// If const constructability is not needed, use the safe version of this
+  /// constructor: [Locale.safe].
   const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
 
+  /// WIP/FIXME: document the constructor properly as it settles down in review.
+  ///
+  /// This constructor matches the unnamed constructor, but does not follow the
+  /// "garbage in, garbage out". This class guarantees standards-compliant
+  /// output if instances are constructed with safe constructors. (The various
+  /// outputs of the Locale instance can be trusted to be syntactically valid
+  /// even if the input is garbage.)
+  ///
+  /// In its current state, this code throws away bad tags. Bad language tags
+  /// become "und" (undefined). Bad region tags are discarded giving a
+  /// regionless Locale.
+  ///
+  /// Other options we could consider is throwing exceptions: it would be nice
+  /// to provide feedback to calling code shoud it want feedback, so that it
+  /// could make its own decision on how to deal with garbage.
+  Locale.safe(String language, { String region }) :
```

Comment:
what is the use case for this?

Reply:
For me, my primary purpose of this pull request is to decide whether we'd like the getters to call toLowerCase() and toUpperCase() - in which case the unnamed constructor becomes case insensitive.

I'll come back to this suggestion/request on Monday, adjusting it in one of two directions:
(a) proposing a change to make the unnamed constructor take care of all of this,
(b) making a Locale.parse proposal to at least get this functionality submitted, following which we could still revisit the question "should the unnamed constructor perhaps be case insensitive?"

Reply:
The purpose of this constructor was to have a constructor that won't ever fail, while producing a "syntactically valid" (as per Unicode LDML specification) Locale instance - i.e. a workaround for the question of "what counts as misuse" and my desire to add some syntax-checking asserts to the existing const nameless constructor.